tests/ttnn/integration_tests/yolov4/custom_preprocessor_d1.py

```diff
@@ -0,0 +1,147 @@
+# SPDX-FileCopyrightText: © 2023 Tenstorrent Inc.
+
+# SPDX-License-Identifier: Apache-2.0
+
+
+from ttnn.model_preprocessing import preprocess_conv2d, fold_batch_norm2d_into_conv2d
+
+from models.experimental.functional_yolov4.reference.downsample1 import DownSample1
+from models.experimental.functional_yolov4.tt.ttnn_downsample1 import TtDownSample1
+
+import ttnn
+import tt_lib
+
+
+def update_ttnn_module_args(ttnn_module_args):
+    ttnn_module_args["use_1d_systolic_array"] = ttnn_module_args.in_channels < 256
+
+
+def custom_preprocessor(device, model, name, ttnn_module_args):
+    parameters = {}
+    if isinstance(model, DownSample1):
+        ttnn_module_args.c1["math_fidelity"] = ttnn.MathFidelity.LoFi
```

Comment:
struct Conv2dConfig {
MathFidelity math_fidelity = MathFidelity::HiFi4;
DataType dtype = DataType::BFLOAT16;
DataType weights_dtype = DataType::BFLOAT16;
bool math_approx_mode_enabled = true;
bool fp32_dest_acc_enabled = false;
bool packer_l1_accum_enabled = false;
string activation = "";
uint32_t input_channels_alignment = 32;
bool deallocate_activation = false;
bool reallocate_halo_output = false;
uint32_t act_block_h_override = 0;
bool reshard_if_not_optimal = false; // if true, override_sharding_config should not be set to true
bool override_sharding_config = false; // if true, reshard_if_not_optimal should not be set to true
bool height_sharding = true; // used only if override_sharding_config is true
std::optional<CoreRangeSet> core_grid = std::nullopt; // used only if override_sharding_config is true
bool transpose_shards = true; // used only if override_sharding_config is true and if height sharding is false
Layout output_layout = Layout::TILE;

Reply:
uint32_t input_channels_alignment = 16. -> shallow conv

Reply:
uint32_t act_block_h_override = 0 -> None